syft/generic/pointers/pointer_tensor.py

```diff
@@ -261,24 +261,47 @@ def create_pointer(
 
         return ptr
 
-    def move(self, destination):
-        kwargs = {"inplace": True, "create_pointer": False}
-        message = TensorCommandMessage.communication(
-            self.id_at_location, self.location.id, [destination.id], kwargs
-        )
-        self.owner.send_msg(message=message, location=self.location)
+    def move(self, destination: AbstractWorker, requires_grad: bool = False):
+        """
+        Will move the remove value from self.location A to destination B
 
-        # Change location of the pointer to point to the new object owner
-        self.location = destination
+        Note a A will keep a copy of his value that he sent to B. This follows the
+        .send() paradigm where the local worker keeps a copy of the value he sends.
 
-        return self
+        Args:
+            destination: the new location of the remote data
+            requires_grad: see send() for details
 
-    def remote_send(self, destination):
+        Returns:
+            A pointer to location
+        """
+        # move to local target is equivalent to doing .get()
+        if self.owner.id == destination.id:
+            return self.get()
+
+        ptr = self.remote_send(destination, requires_grad=requires_grad)
+
+        # We make the pointer point at the remote value. As the id doesn't change,
+        # we don't update ptr.id_at_location
+        # Note that you have now 2 pointers on different locations pointing to the
+        # same tensor.
```

Comment:
This could definitely cause issues :)
Not to be fixed in this PR but maybe we can create an issue or keep track of this in another way

Reply:
Will do:  #3217